src/commands/install.yml

```diff
@@ -18,39 +18,62 @@ steps:
   - run:
       name: "Install SecretHub CLI"
       shell: << parameters.shell >>
+      environment:
+        SECRETHUB_CLI_VERSION: << parameters.version >>
       command: |
 
-        # If the version is not pinned then check if SecretHub is already installed.
-        if [ "<< parameters.version >>" = "latest" ] && command -v secrethub >/dev/null 2>&1; then
-            exit 0
-        fi
+        set -e
+
+        # Colors
+        NO_COLOR="\033[0m"
+        OK_COLOR="\033[32;01m"
+        ERROR_COLOR="\033[31;01m"
+        WARN_COLOR="\033[33;01m"
 
-        # Get the latest version.
-        VERSION="<< parameters.version >>"
-        if [ "$VERSION" = "latest" ]; then
-            VERSION=$(curl -sSfL "https://api.github.com/repos/secrethub/secrethub-cli/releases/latest" | grep tag_name | awk -F\" '{ print substr($4,2) }')
+        # Detect Architecture
+        ARCH=amd64
+        if [ $(getconf LONG_BIT) = 32 ]; then
+            ARCH=386
         fi
 
-        # Get the current platform.
-        if uname -a | grep Darwin > /dev/null 2>&1; then
-            PLATFORM=darwin
+        # Detect OS
+        UNAME=$(uname)
+        if [ "$UNAME" = "Darwin" ]; then
+          OS=darwin
+        elif [ "$UNAME" = "Linux" ]; then
+            OS=linux
         else
-            PLATFORM=linux
+            echo -e "${ERROR_COLOR}Cannot determine OS type. Exiting...${NO_COLOR}"
+            exit;
+        fi
+
+        # Make sure we have root priviliges.
+        SUDO=""
+        if [ $(id -u) -ne 0 ]; then
+            if ! [ $(command -v sudo) ]; then
```

Comment:
Should we remove this check as well now? You've already removed the `sudo -v` command below.

Reply:
This still has a purpose: checking if you even have `sudo`.

Reply:
Ah I get it know. If you're already root, we don't use `sudo`, as the `SUDO` variable is set to the empty string. So in that case, you also don't need the `sudo` command. Thanks